src/integrationTest/java/com/mongodb/hibernate/query/select/AbstractSelectionQueryIntegrationTests.java

```diff
@@ -0,0 +1,142 @@
+import org.hibernate.query.SelectionQuery;
+import org.hibernate.testing.orm.junit.ServiceRegistryScope;
+import org.hibernate.testing.orm.junit.ServiceRegistryScopeAware;
+import org.hibernate.testing.orm.junit.SessionFactory;
+import org.hibernate.testing.orm.junit.SessionFactoryScope;
+import org.hibernate.testing.orm.junit.SessionFactoryScopeAware;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@SessionFactory(exportSchema = false)
+@ExtendWith(MongoExtension.class)
+abstract class AbstractSelectionQueryIntegrationTests implements SessionFactoryScopeAware, ServiceRegistryScopeAware {
+
+    private SessionFactoryScope sessionFactoryScope;
+
+    private TestCommandListener testCommandListener;
+
+    protected SessionFactoryScope getSessionFactoryScope() {
+        return sessionFactoryScope;
+    }
+
+    protected TestCommandListener getTestCommandListener() {
+        return testCommandListener;
+    }
+
+    @Override
+    public void injectSessionFactoryScope(SessionFactoryScope sessionFactoryScope) {
+        this.sessionFactoryScope = sessionFactoryScope;
+    }
+
+    @Override
+    public void injectServiceRegistryScope(ServiceRegistryScope serviceRegistryScope) {
+        this.testCommandListener = serviceRegistryScope.getRegistry().requireService(TestCommandListener.class);
+    }
+
+    <T> void assertSelectionQuery(
+            String hql,
+            Class<T> resultType,
+            Consumer<SelectionQuery<T>> queryPostProcessor,
+            String expectedMql,
+            List<T> expectedResultList) {
+        assertSelectionQuery(hql, resultType, queryPostProcessor, expectedMql, resultList -> assertThat(resultList)
+                .usingRecursiveFieldByFieldElementComparator()
+                .containsExactlyElementsOf(expectedResultList));
+    }
+
+    <T> void assertSelectionQuery(String hql, Class<T> resultType, String expectedMql, List<T> expectedResultList) {
+        assertSelectionQuery(hql, resultType, null, expectedMql, expectedResultList);
+    }
+
+    <T> void assertSelectionQuery(
+            String hql,
+            Class<T> resultType,
+            Consumer<SelectionQuery<T>> queryPostProcessor,
+            String expectedMql,
+            Consumer<List<T>> resultListVerifier) {
+        sessionFactoryScope.inTransaction(session -> {
+            var selectionQuery = session.createSelectionQuery(hql, resultType);
+            if (queryPostProcessor != null) {
+                queryPostProcessor.accept(selectionQuery);
+            }
+            var resultList = selectionQuery.getResultList();
+
+            assertActualCommand(BsonDocument.parse(expectedMql));
+
+            resultListVerifier.accept(resultList);
+        });
+    }
+
+    <T> void assertSelectionQuery(
+            String hql, Class<T> resultType, String expectedMql, Consumer<List<T>> resultListVerifier) {
+        assertSelectionQuery(hql, resultType, null, expectedMql, resultListVerifier);
+    }
```

Comment:
I suggest format the parameters to match the pattern used in the rest of the class for consistency.

```suggestion
    <T> void assertSelectionQuery(
            String hql,
            Class<T> resultType,
            String expectedMql,
            Consumer<List<T>> resultListVerifier) {
        assertSelectionQuery(hql, resultType, null, expectedMql, resultListVerifier);
    }
```

Reply:
The formatting is done automatically by `spotlessApply`. There used to be a `static-check` bug (`spotlessApply` is applied automatically) which hides the issue. Now I updated the PR to include the evergreen bug fix, now the above two `Commit suggestion` would result in evergreen building failure.

Had to revert back to satisfy `spotless`.